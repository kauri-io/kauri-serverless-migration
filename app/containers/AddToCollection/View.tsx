import AlertView from '../../components/Modal/AlertView'
import AddToCollectionModalContent from './AddToCollectionModalContent'
import { Query } from 'react-apollo'
import { path } from 'ramda'
import { ICollection } from './CollectionsContent'
import { ISection } from './SectionsContent'
import Loading from '../../components/Loading'
import { ErrorMessage } from '../../lib/with-apollo-error'
import { Label } from '../..//components/Typography'
import { getCollectionsForUser as query } from '../../queries/Collection'
import {
    getCollectionsForUser,
    getCollectionsForUserVariables,
} from '../../queries/__generated__/getCollectionsForUser'
import { useState } from 'react'
import { addResourceToCollectionAction } from './Module'
import { ResourceTypeInput } from '../../__generated__/globalTypes'

interface IChosen {
    chosenCollection: ICollection | null
    chosenSection: ISection | null
}

interface IProps {
    closeModalAction: () => void
    userId: string | null
    routeChangeAction: (route: string) => void
    resourceId: string
    type: ResourceTypeInput
    addResourceToCollectionAction: typeof addResourceToCollectionAction
}

const AddToCollectionComponent: React.FunctionComponent<IProps> = ({
    closeModalAction,
    userId,
    routeChangeAction,
    resourceId,
    type,
    addResourceToCollectionAction,
}) => {
    const [state, setState] = useState<IChosen>({
        chosenCollection: null,
        chosenSection: null,
    })

    return (
        <Query<getCollectionsForUser, getCollectionsForUserVariables>
            query={query}
            variables={{ filter: { ownerIdEquals: userId } }}
        >
            {props => {
                if (props.loading) {
                    return <Loading />
                }
                if (props.error) {
                    return (
                        <ErrorMessage
                            data={{ error: { message: props.error.message } }}
                        />
                    )
                }

                if (
                    props.data &&
                    props.data.searchCollections &&
                    Array.isArray(props.data.searchCollections.content)
                ) {
                    const collections = path<ICollection[]>([
                        'searchCollections',
                        'content',
                    ])(props.data)

                    const collectionsThatDoNotHaveTheChosenArticleId =
                        (collections &&
                            Array.isArray(collections) &&
                            collections.filter(collection => {
                                // Filter out collections that have the resourceId already
                                if (
                                    collection.sections.find(
                                        section =>
                                            !!section.resources.find(
                                                resource =>
                                                    resource.id === resourceId
                                            )
                                    )
                                ) {
                                    return null
                                } else {
                                    return collection
                                }
                            })) ||
                        []

                    const articleAlreadyInAllCollections =
                        collectionsThatDoNotHaveTheChosenArticleId &&
                        collectionsThatDoNotHaveTheChosenArticleId.length === 0

                    return Array.isArray(collections) &&
                        collections.length > 0 ? (
                        <AlertView
                            closeModalAction={closeModalAction}
                            confirmButtonText={
                                articleAlreadyInAllCollections
                                    ? 'CREATE NEW COLLECTION'
                                    : 'CONFIRM'
                            }
                            confirmButtonAction={() => {
                                if (articleAlreadyInAllCollections) {
                                    closeModalAction()
                                    routeChangeAction(
                                        `/create-collection?resourceId=${resourceId}&type=${type}`
                                    )
                                }

                                if (
                                    state.chosenCollection &&
                                    state.chosenSection
                                ) {
                                    const chosenSectionid =
                                        state.chosenSection.id
                                    const chosenCollectionSections = state.chosenCollection.sections.find(
                                        section =>
                                            section.id === chosenSectionid
                                    )

                                    // Always insert article at the end
                                    const position = chosenCollectionSections
                                        ? Array.isArray(
                                              chosenCollectionSections.resources
                                          )
                                            ? chosenCollectionSections.resources
                                                  .length
                                            : 0
                                        : 0

                                    addResourceToCollectionAction(
                                        {
                                            id: state.chosenCollection.id,
                                            position,
                                            resourceId: {
                                                id: resourceId,
                                                type,
                                            },
                                            sectionId: state.chosenSection
                                                ? state.chosenSection.id
                                                : '',
                                            closeModalAction,
                                            routeChangeAction,
                                        },
                                        closeModalAction
                                    )
                                }
                            }}
                            content={
                                <AddToCollectionModalContent
                                    changeToPrefilledArticleCreateCollectionRoute={() => {
                                        closeModalAction()
                                        routeChangeAction(
                                            `/create-collection?resourceId=${resourceId}&type=${type}`
                                        )
                                    }}
                                    articleAlreadyInAllCollections={
                                        !!articleAlreadyInAllCollections
                                    }
                                    collectionsThatDoNotHaveTheChosenArticleId={
                                        collectionsThatDoNotHaveTheChosenArticleId
                                    }
                                    parentState={state}
                                    setCollection={({ chosenCollection }) =>
                                        setState({
                                            chosenCollection,
                                            chosenSection: null,
                                        })
                                    }
                                    setSection={({ chosenSection }) =>
                                        setState({ ...state, chosenSection })
                                    }
                                />
                            }
                            title={'Add to Collection'}
                        />
                    ) : (
                        <AlertView
                            title={'Create new collection'}
                            closeModalAction={closeModalAction}
                            confirmButtonAction={() => {
                                closeModalAction()
                                routeChangeAction(
                                    `/create-collection?resourceId=${resourceId}&type=${type}`
                                )
                            }}
                            confirmButtonText={'Create new collection'}
                            content={
                                <section>
                                    <Label>
                                        You don't have any collections.
                                    </Label>
                                </section>
                            }
                        />
                    )
                }
            }}
        </Query>
    )
}

export default AddToCollectionComponent
