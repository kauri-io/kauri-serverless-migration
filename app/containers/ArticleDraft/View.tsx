import DraftArticleHeader from "./DraftArticleHeader";
import DraftArticleContent from "./DraftArticleContent";
import { IOption } from "../../common/PublishingSelector";

interface IProps {
  data: {
    getArticle: any;
  };
  deleteDraftArticleAction: (
    { id, version }: { id: string; version: number }
  ) => void;
  openModalAction: (children?: any) => void;
  closeModalAction: () => void;
  publishArticleAction: any;
  userId: string;
  routeChangeAction: (route: string) => void;
  communities: Array<{ community: IOption }>;
}
export default ({
  data: { getArticle },
  deleteDraftArticleAction,
  closeModalAction,
  publishArticleAction,
  openModalAction,
  userId,
  routeChangeAction,
  communities,
}: IProps) => (
  <>
    <DraftArticleHeader
      {...getArticle}
      communities={communities}
      userId={userId}
      routeChangeAction={routeChangeAction}
      publishArticleAction={publishArticleAction}
      openModalAction={openModalAction}
      closeModalAction={closeModalAction}
    />
    <DraftArticleContent
      {...getArticle}
      closeModalAction={closeModalAction}
      openModalAction={openModalAction}
      deleteDraftArticleAction={deleteDraftArticleAction}
    />
  </>
);
