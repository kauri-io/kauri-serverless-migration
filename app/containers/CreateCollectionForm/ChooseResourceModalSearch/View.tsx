import React from "react";
import { IResult } from "../../../../../kauri-components/components/Search/QuickSearch";
import styled from "styled-components";
import { Subject } from "rxjs/Subject";
import { Subscription } from "rxjs/Subscription";
import { Observable } from "rxjs/Observable";

const SearchSVG = () => (
  <div className="certain-category-icon">
    <svg width="18" height="18" viewBox="0 0 24 24" fill="#1E2428">
      <path
        fillRule="evenodd"
        clipRule="evenodd"
        d="M11 4C7.13401 4 4 7.13401 4 11C4 14.866 7.13401 18 11 18C14.866 18 18 14.866 18 11C18 7.13401 14.866 4 11 4ZM2 11C2 6.02944 6.02944 2 11 2C15.9706 2 20 6.02944 20 11C20 15.9706 15.9706 20 11 20C6.02944 20 2 15.9706 2 11Z"
        fill="#1E2428"
      />
      <path
        fillRule="evenodd"
        clipRule="evenodd"
        d="M15.9429 15.9429C16.3334 15.5524 16.9666 15.5524 17.3571 15.9429L21.7071 20.2929C22.0976 20.6834 22.0976 21.3166 21.7071 21.7071C21.3166 22.0977 20.6834 22.0977 20.2929 21.7071L15.9429 17.3571C15.5524 16.9666 15.5524 16.3334 15.9429 15.9429Z"
        fill="#1E2428"
      />
    </svg>
  </div>
);

const SearchInput = styled.input`
  border: 1px solid ${props => props.theme.colors.bgPrimary};
  border-radius: 4px;
  background: transparent;
  height: 40px;
  outline: none;
  padding: 0 ${props => props.theme.space[1]}px;
`;

const SearchWrapper = styled<{ type: string }, "div">("div")`
  width: ${props => (props.type === "article" ? "330" : "357")}px;
  display: grid;
  position: relative;

  &:hover {
    & > input {
      border: 1px solid #209b86;
    }
  }

  & > .certain-category-icon {
    position: absolute;
    top: 11px;
    right: 9px;
  }
`;

interface ISearch {
  value: string;
  type?: string;
}

const handleSearch$: Subject<ISearch> = new Subject();

interface IDataSource {
  results: IResult[];
}

const EmptyData: IDataSource = {
  results: [],
};

interface IState {
  dataSource: IDataSource;
  sub: Subscription | null;
  value: string;
}

interface IProps {
  query: any;
  routeChangeAction: (route: string) => void;
  category: string | null;
  changeTab: (index: number) => void;
  type: "article" | "collection";
}

const queryAllArticles = (refetchQuery: any, text: string) =>
  Observable.fromPromise(
    refetchQuery({
      page: 0,
      size: 8,
      text: text === "" ? null : text,
    })
  );

const queryAllCollections = (refetchQuery: any, text: string) =>
  Observable.fromPromise(
    refetchQuery({
      filter: {
        fullText: text === "" ? null : text,
      },
      page: 0,
      size: 8,
    })
  );

class Complete extends React.Component<IProps, IState> {
  constructor(props: IProps) {
    super(props);
    this.state = {
      dataSource: EmptyData,
      sub: null,
      value: "",
    };
  }

  componentDidMount() {
    const sub = handleSearch$
      .debounceTime(200)
      .flatMap(() =>
        this.props.type === "article"
          ? queryAllArticles(this.props.query.refetch, this.state.value)
          : queryAllCollections(this.props.query.refetch, this.state.value)
      )
      .do(() => this.props.changeTab(1))
      .map(({ data }: any) => ({
        results:
          (data.searchCollections && data.searchCollections.content) || [],
      }))
      .subscribe(
        dataSource => {
          // console.log(dataSource);
          this.setState({ ...this.state, dataSource });
        },
        (err: string) => console.log(err)
      );
    this.setState({ ...this.state, sub });
  }

  componentWillUnmount() {
    if (this.state.sub) {
      this.state.sub.unsubscribe();
    }
  }

  fetchResults = (value: string) => {
    this.setState({ value });
    handleSearch$.next({ value });
  };

  render() {
    return (
      <SearchWrapper type={this.props.type} className="global-search-wrapper">
        <SearchInput
          onChange={({ target: { value } }) => this.fetchResults(value)}
        />
        <SearchSVG />
      </SearchWrapper>
    );
  }
}

export default Complete;
