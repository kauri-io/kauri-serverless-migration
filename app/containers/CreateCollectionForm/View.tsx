// @flow
import * as React from "react";
import styled from "styled-components";
import { space, bg } from "styled-system";
import { Form, Field, FieldArray } from "formik";
import Stack from "stack-styled";
import R from "ramda";
import ActionsSection from "../../../../kauri-components/components/Section/ActionsSection";
import PrimaryHeaderSection from "../../../../kauri-components/components/Section/PrimaryHeaderSection";
import CardContentSection from "../../../../kauri-components/components/Section/CardContentSection";
import StatisticsContainer from "../../../../kauri-components/components/PublicProfile/StatisticsContainer";
import UserAvatar from "../../../../kauri-components/components/UserAvatar";
import { Label } from "../../../../kauri-components/components/Typography";
import Input from "../../../../kauri-components/components/Input/Input";
import PrimaryButton from "../../../../kauri-components/components/Button/PrimaryButton";
import TertiaryButton from "../../../../kauri-components/components/Button/TertiaryButton";
import ArticleCard from "../../connections/ArticleCard";
import CollectionCard from "../../connections/CollectionCard";
import setImageUploader from "../../common/ImageUploader";
import showFormValidationErrors from "../../../lib/show-form-validation-errors";
import ChooseArticleModal from "./ChooseArticleModal";
import ChooseCollectionModal from "./ChooseCollectionModal";
import SectionOptions from "./SectionOptions";
// import AddTagButton from '../../../../kauri-components/components/Button/AddTagButton'
// import AddMemberButton from '../../../../kauri-components/components/Button/AddMemberButton'
import TagSelector from "../../common/TagSelector";
import { DragDropContext, Draggable, Droppable } from "react-beautiful-dnd";
import handleDragEnd from "./handleDragEnd";

import type { FormState } from "./index";
import type { ShowNotificationPayload } from "../../../lib/Module";

const emptySection: SectionDTO = {
  name: "",
  description: undefined,
  resourcesId: [],
};
const RemoveIcon = () => (
  <img src="https://png.icons8.com/windows/50/000000/delete-sign.png" />
);

const Section = styled.section`
  display: flex;
  flex-direction: column;
`;

const ResourceSection = styled.section`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-end;
  ${space};
  > button:last-child {
    margin-top: ${props => props.theme.space[1]}px;
  }
`;

const SectionSection = styled.section`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  :not(:first-child) {
    ${space};
  }
  > :nth-child(2) {
    margin-bottom: ${props => props.theme.space[2]}px;
  }
  > button:nth-child(4) {
    margin-top: ${props => props.theme.space[2]}px;
  }
`;

const ContentSection = styled.section`
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: ${props => props.theme.paddingTop} ${props => props.theme.padding};
  min-height: calc(100vh - 270px);
  ${bg};
`;

const CreateCollectionDetails = styled.div`
  display: flex;
  flex-direction: column;
  > :not(:last-child) {
    ${space};
  }
`;

const AddAnotherSectionContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  ${space};
`;
const CreateCollectionActionsPlaceHolder = styled.div`
  display: flex;
  mix-blend-mode: normal;
  opacity: 0.2;
  cursor: initial !important;
  > * {
    ${space};
    cursor: initial !important;
  }
`;

const CreateCollectionMetaDetails = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  > * {
    ${space};
  }
`;

const CreateCollectionCuratorDetails = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  ${space};
  > :first-child {
    margin-bottom: ${props => props.theme.space[3]}px;
  }
  > :nth-child(2) {
    margin-bottom: ${props => props.theme.space[1]}px;
  }
`;

const CreateCollectionCurators = styled.div`
  display: flex;
  align-items: center;
`;

const UploadIcon = () => (
  <img src="https://png.icons8.com/color/50/000000/upload.png" />
);

const DisplayFormikState = props => (
  <div style={{ margin: "1rem 0", background: "#f6f8fa", padding: ".5rem" }}>
    <strong>Injected Formik props (the form's state)</strong>
    <div>
      <code>errors:</code> {JSON.stringify(props.errors, null, 2)}
    </div>
    <div>
      <code>values:</code> {JSON.stringify(props.values, null, 2)}
    </div>
    <div>
      <code>isSubmitting:</code> {JSON.stringify(props.isSubmitting, null, 2)}
    </div>
  </div>
);

const ErrorMessageRenderer = styled.h2`
  color: #ffffff !important;
`;

const ShareIcon = () => (
  <svg role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512">
    <path
      fill="#0BA986"
      d="M352 320c-22.608 0-43.387 7.819-59.79 20.895l-102.486-64.054a96.551 96.551 0 0 0 0-41.683l102.486-64.054C308.613 184.181 329.392 192 352 192c53.019 0 96-42.981 96-96S405.019 0 352 0s-96 42.981-96 96c0 7.158.79 14.13 2.276 20.841L155.79 180.895C139.387 167.819 118.608 160 96 160c-53.019 0-96 42.981-96 96s42.981 96 96 96c22.608 0 43.387-7.819 59.79-20.895l102.486 64.054A96.301 96.301 0 0 0 256 416c0 53.019 42.981 96 96 96s96-42.981 96-96-42.981-96-96-96z"
      className=""
    />
  </svg>
);

const DraggableResourceContainer = styled.div`
  :hover {
    > :first-child {
      box-shadow: 0 0 0 2px ${props => props.theme.hoverTextColor};
    }
  }
`;

const handleBackgroundSetFormField = setFieldValue => () =>
  setImageUploader(payload => {
    setFieldValue("background", payload.background.background);
  }, "background");

const renderResourceSection = (
  index,
  arrayHelpers,
  section,
  values,
  mappingKey,
  provided
) => (resource, resourceIndex) => (
  <ResourceSection key={resourceIndex} mt={3}>
    {R.path(
      ["sections", index, mappingKey, resourceIndex, "version"],
      values
    ) ? (
      <Draggable
        index={resourceIndex}
        draggableId={`${R.path(
          ["sections", index, mappingKey, resourceIndex, "id"],
          values
        )}-${R.path(
          ["sections", index, mappingKey, resourceIndex, "version"],
          values
        )}`}
      >
        {provided => (
          <DraggableResourceContainer
            {...provided.draggableProps}
            {...provided.dragHandleProps}
            innerRef={provided.innerRef}
            id="article-card"
          >
            <ArticleCard
              id={R.path(
                ["sections", index, mappingKey, resourceIndex, "id"],
                values
              )}
              version={parseInt(
                R.path(
                  ["sections", index, mappingKey, resourceIndex, "version"],
                  values
                )
              )}
            />
            {provided.placeholder}
          </DraggableResourceContainer>
        )}
      </Draggable>
    ) : (
      R.path(["sections", index, mappingKey, resourceIndex], values) && (
        <Draggable
          index={resourceIndex}
          draggableId={`${R.path(
            ["sections", index, mappingKey, resourceIndex, "id"],
            values
          )}`}
        >
          {provided => (
            <DraggableResourceContainer
              {...provided.draggableProps}
              {...provided.dragHandleProps}
              innerRef={provided.innerRef}
              id="collection-card"
            >
              <CollectionCard
                id={R.path(
                  ["sections", index, mappingKey, resourceIndex, "id"],
                  values
                )}
              />
              {provided.placeholder}
            </DraggableResourceContainer>
          )}
        </Draggable>
      )
    )}
    <TertiaryButton
      color="primaryTextColor"
      icon={<RemoveIcon />}
      onClick={() =>
        arrayHelpers.form.setFieldValue(
          `sections[${index}][${mappingKey}]`,
          Array.isArray(section[mappingKey]) &&
            (!resourceIndex
              ? section[mappingKey].length > 1
                ? section[mappingKey].splice(1)
                : []
              : R.remove(resourceIndex, resourceIndex, section[mappingKey]))
        )
      } // Remove current resource index
    >
      {`Remove ${resource.type}`}
    </TertiaryButton>
  </ResourceSection>
);

type Props = {
  id?: string,
  userId: string,
  touched: {
    name: boolean,
    description: boolean,
  },
  errors: {
    name: ?string,
    description: ?string,
  },
  values: FormState,
  isSubmitting: boolean,
  setFieldValue: (string, any) => void,
  validateForm: () => Promise<any>,
  showNotificationAction: ShowNotificationPayload => void,
  routeChangeAction: string => void,
  data?: { getCollection?: ?CollectionDTO },
  openModalAction: ({ children: React.Node }) => void,
  closeModalAction: () => void,
  userId: string,
  username: string,
  userAvatar: string,
  isLoggedIn: boolean,
};

const BackIcon = styled.div`
  width: 10px !important;
  height: 14px !important;
  border-top: 8px solid transparent;
  border-bottom: 8px solid transparent;
  border-right: 10px solid ${props => props.theme.colors["primary"]};
`;

export default ({
  id,
  touched,
  errors,
  values,
  isSubmitting,
  setFieldValue,
  validateForm,
  showNotificationAction,
  routeChangeAction,
  data,
  openModalAction,
  closeModalAction,
  username,
  userId,
  userAvatar,
}: Props) => (
  <Section>
    <Form>
      <ActionsSection
        bg={
          (typeof values.background === "string" && "transparent") ||
          "bgPrimary"
        }
      >
        <Stack alignItems={["", "center"]}>
          <TertiaryButton
            onClick={() => routeChangeAction("back")}
            icon={<BackIcon />}
          >
            Cancel Collection
          </TertiaryButton>
        </Stack>
        <Stack alignItems={["", "center"]} justifyContent={["", "center"]}>
          <TertiaryButton
            icon={<UploadIcon />}
            handleClick={handleBackgroundSetFormField(setFieldValue)}
          >
            Background Image
          </TertiaryButton>
        </Stack>
        <Stack alignItems={["", "center"]} justifyContent={["", "end"]}>
          <PrimaryButton
            disabled={isSubmitting}
            type="submit"
            onClick={() =>
              showFormValidationErrors(validateForm, showNotificationAction)
            }
          >
            {data ? "Update Collection" : "Create Collection"}
          </PrimaryButton>
        </Stack>
      </ActionsSection>

      <PrimaryHeaderSection backgroundURL={values.background}>
        <CreateCollectionDetails mb={2}>
          <Label color="white">Collection</Label>
          <Field
            type="text"
            name="name"
            render={({ field }) => (
              <Input
                {...field}
                type="text"
                placeHolder="Add collection title"
                fontSize={7}
              />
            )}
          />
          {/* <ErrorMessage name='name' render={(message: string) => <ErrorMessageRenderer>{message}</ErrorMessageRenderer>} /> */}
          <Field
            type="text"
            name="description"
            render={({ field }) => (
              <Input
                {...field}
                type="text"
                placeHolder="Add description"
                fontSize={4}
              />
            )}
          />

          <FieldArray
            name="tags"
            render={arrayHelpers => (
              <TagSelector
                updateTags={tags =>
                  arrayHelpers.form.setFieldValue("tags", tags)
                }
                tags={values.tags || []}
              />
            )}
          />
          {/* <ErrorMessage name='description' render={(message: string) => <ErrorMessageRenderer>{message}</ErrorMessageRenderer>} /> */}

          {/* TODO: WAIT FOR BACKEND */}
          {/* <AddTagButton color='white' /> */}
          <CreateCollectionActionsPlaceHolder mr={3}>
            {/* <PrimaryButton>Follow Collection</PrimaryButton> */}
            {/* <TertiaryButton>Up vote</TertiaryButton> */}
            {/* <TertiaryButton icon={<ShareIcon />}>Share</TertiaryButton> */}
          </CreateCollectionActionsPlaceHolder>
        </CreateCollectionDetails>
        <Stack alignItems={["", "center"]} justifyContent={["", "end"]}>
          <CreateCollectionMetaDetails mb={4}>
            <CreateCollectionCuratorDetails mr={4} mb={2}>
              <StatisticsContainer
                pageType="CreateCollectionPage"
                statistics={[
                  {
                    name: "Articles",
                    count: R.pipe(
                      R.map(({ resourcesId }) => resourcesId),
                      R.reduce((current, next) => {
                        const articlesInSection = next.filter(
                          ({ type }) => type.toLowerCase() === "article"
                        );
                        if (articlesInSection) {
                          return current + articlesInSection.length;
                        }
                        return current;
                      }, 0)
                    )(values.sections),
                  },
                  {
                    name: "Collections",
                    count: R.pipe(
                      R.map(({ resourcesId }) => resourcesId),
                      R.reduce((current, next) => {
                        const collectionsInSection = next.filter(
                          ({ type }) => type.toLowerCase() === "collection"
                        );
                        if (collectionsInSection) {
                          return current + collectionsInSection.length;
                        }
                        return current;
                      }, 0)
                    )(values.sections),
                  },
                ]}
              />
              <Label color="white">Curator</Label>
              <CreateCollectionCurators>
                <UserAvatar
                  variant="white"
                  fullWidth
                  username={username}
                  userId={userId}
                  avatar={userAvatar}
                />
                {/* <AddMemberButton /> */}
              </CreateCollectionCurators>
            </CreateCollectionCuratorDetails>
          </CreateCollectionMetaDetails>
        </Stack>
      </PrimaryHeaderSection>

      <ContentSection bg="tertiaryBackgroundColor">
        <FieldArray
          name="sections"
          render={arrayHelpers => (
            <React.Fragment>
              {/* {console.log(arrayHelpers)} */}
              {values.sections &&
                values.sections.length > 0 &&
                values.sections.map((section: SectionDTO, index) => (
                  <SectionSection key={index} mt={4}>
                    <Field
                      type="text"
                      name={`sections.${index}.name`}
                      render={({ field }) => (
                        <Input
                          {...field}
                          type="text"
                          placeHolder="Add Section Name"
                          fontSize={5}
                          fontWeight={500}
                          color={"primaryTextColor"}
                          textAlign={"center"}
                        />
                      )}
                    />
                    <Field
                      type="text"
                      name={`sections.${index}.description`}
                      render={({ field }) => (
                        <Input
                          {...field}
                          type="text"
                          placeHolder="Add Section Description"
                          fontSize={2}
                          fontWeight={300}
                          color={"primaryTextColor"}
                          textAlign={"center"}
                        />
                      )}
                    />

                    <DragDropContext
                      onDragEnd={handleDragEnd(arrayHelpers, index, values)}
                    >
                      <Droppable
                        direction={"horizontal"}
                        droppableId={section.id || "0"}
                      >
                        {provided => (
                          <CardContentSection
                            {...provided.droppableProps}
                            innerRef={provided.innerRef}
                          >
                            {/* Section id */}
                            {section.resourcesId &&
                              Array.isArray(section.resourcesId) &&
                              section.resourcesId.map(
                                renderResourceSection(
                                  index,
                                  arrayHelpers,
                                  section,
                                  values,
                                  "resourcesId"
                                )
                              )}
                          </CardContentSection>
                        )}
                      </Droppable>
                    </DragDropContext>

                    <SectionOptions
                      currentSectionIndex={index}
                      previousSectionHasArticles={R.pipe(
                        R.path([
                          "sections",
                          index > 0 ? index : 0,
                          "resourcesId",
                        ]),
                        R.defaultTo([]),
                        resourcesId => resourcesId.length,
                        Boolean
                      )(values)}
                      addNewSection={() => arrayHelpers.push(emptySection)}
                      removeSection={() => arrayHelpers.remove(index)}
                      chooseArticle={() =>
                        openModalAction({
                          children: (
                            <ChooseArticleModal
                              allOtherChosenArticles={values.sections.filter(
                                (section, sectionIndex) =>
                                  index !== sectionIndex
                              )}
                              chosenArticles={R.pipe(
                                R.path(["sections", index, "resourcesId"]),
                                R.filter(
                                  ({ type }) => type.toLowerCase() === "article"
                                )
                              )(values)}
                              closeModalAction={() => closeModalAction()}
                              confirmModal={chosenArticles =>
                                arrayHelpers.form.setFieldValue(
                                  `sections[${index}].resourcesId`,
                                  R.pipe(
                                    R.path(["sections", index, "resourcesId"]),
                                    R.filter(
                                      ({ type }) =>
                                        type.toLowerCase() === "collection"
                                    ),
                                    R.concat(
                                      chosenArticles.map(article => ({
                                        ...article,
                                        type: "ARTICLE",
                                      }))
                                    )
                                  )(values)
                                )
                              }
                            />
                          ),
                        })
                      }
                      chooseCollection={() =>
                        openModalAction({
                          children: (
                            <ChooseCollectionModal
                              currentCollectionIdIfUpdating={id}
                              allOtherChosenCollections={values.sections.filter(
                                (section, sectionIndex) =>
                                  index !== sectionIndex
                              )}
                              chosenCollections={R.pipe(
                                R.path(["sections", index, "resourcesId"]),
                                R.filter(
                                  ({ type }) =>
                                    type.toLowerCase() === "collection"
                                )
                              )(values)}
                              closeModalAction={() => closeModalAction()}
                              confirmModal={chosenCollections =>
                                arrayHelpers.form.setFieldValue(
                                  `sections[${index}].resourcesId`,
                                  R.pipe(
                                    R.path(["sections", index, "resourcesId"]),
                                    R.filter(
                                      ({ type }) =>
                                        type.toLowerCase() === "article"
                                    ),
                                    R.concat(
                                      chosenCollections.map(collection => ({
                                        ...collection,
                                        type: "COLLECTION",
                                      }))
                                    )
                                  )(values)
                                )
                              }
                            />
                          ),
                        })
                      }
                    />
                  </SectionSection>
                ))}
            </React.Fragment>
          )}
        />

        {process.env.NODE_ENV !== "production" && (
          <DisplayFormikState
            touched={touched}
            errors={errors}
            values={values}
            isSubmitting={isSubmitting}
          />
        )}
      </ContentSection>
    </Form>
  </Section>
);
