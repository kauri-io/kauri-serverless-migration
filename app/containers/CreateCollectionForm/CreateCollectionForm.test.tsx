import View from './View'
import { shallow } from 'enzyme'

describe('containers/CreateCollectionForm/View', () => {
    it('should match the snapshot', () => {
        const props = {
            communities: [],
            id: '123',
            touched: {
                name: true,
                description: true,
            },
            errors: {},
            values: {
                name: 'test name',
                sections: [],
                tags: [],
            },
            isSubmitting: false,
            setFieldValue: jest.fn(),
            validateForm: jest.fn(),
            showNotificationAction: jest.fn(),
            routeChangeAction: jest.fn(),
            openModalAction: jest.fn(),
            editCollectionAction: jest.fn(),
            createCollectionAction: jest.fn(),
            closeModalAction: jest.fn(),
            userId: '234',
            username: '345',
            userAvatar: 'avatar url',
            isLoggedIn: true,
            query: {
                articleId: '789',
                version: '2',
            },
            isValidating: false,
            submitCount: 0,
            setSubmitting: jest.fn(),
            setStatus: jest.fn(),
            setError: jest.fn(),
            setErrors: jest.fn(),
            setTouched: jest.fn(),
            setValues: jest.fn(),
            setFieldError: jest.fn(),
            setFieldTouched: jest.fn(),
            validateField: jest.fn(),
            resetForm: jest.fn(),
            submitForm: jest.fn(),
            setFormikState: jest.fn(),
            handleSubmit: jest.fn(),
            handleReset: jest.fn(),
            handleBlur: jest.fn(),
            handleChange: jest.fn(),
            dirty: false,
            isValid: true,
            initialValues: {
                name: 'test name',
                sections: [],
                tags: [],
            },
            registerField: jest.fn(),
            unregisterField: jest.fn(),
        }
        const wrapper = shallow(<View {...props} />)
        expect(wrapper).toMatchSnapshot()
    })
})
