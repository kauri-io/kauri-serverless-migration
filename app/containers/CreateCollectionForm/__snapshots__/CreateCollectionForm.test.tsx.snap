// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/CreateCollectionForm/View should match the snapshot 1`] = `
<View__Section>
  <Form>
    <ActionsSection
      bg="bgPrimary"
    >
      <Stack
        alignItems={
          Array [
            "",
            "center",
          ]
        }
        gap={2}
      >
        <TertiaryButtonComponent
          icon={<ForwardRef(View__BackIcon) />}
          onClick={[Function]}
        >
          Cancel Collection
        </TertiaryButtonComponent>
      </Stack>
      <Stack
        alignItems={
          Array [
            "",
            "center",
          ]
        }
        gap={2}
        justifyContent={
          Array [
            "",
            "center",
          ]
        }
      >
        <TertiaryButtonComponent
          handleClick={[Function]}
          icon={<UploadIcon />}
        >
          Background Image
        </TertiaryButtonComponent>
      </Stack>
      <Stack
        alignItems={
          Array [
            "",
            "center",
          ]
        }
        gap={2}
        justifyContent={
          Array [
            "",
            "end",
          ]
        }
      >
        <PrimaryButtonComponent
          disabled={false}
          onClick={[Function]}
          type="submit"
        >
          Create Collection
        </PrimaryButtonComponent>
      </Stack>
    </ActionsSection>
    <PrimaryHeaderSection>
      <View__CreateCollectionDetails>
        <Typography__component
          color="white"
        >
          Collection
        </Typography__component>
        <FormikConnect(FieldInner)
          name="name"
          render={[Function]}
          type="text"
        />
        <FormikConnect(FieldInner)
          name="description"
          render={[Function]}
          type="text"
        />
        <FormikConnect(FieldArrayInner)
          name="tags"
          render={[Function]}
        />
        <View__CreateCollectionActionsPlaceHolder />
      </View__CreateCollectionDetails>
      <Stack
        alignItems={
          Array [
            "",
            "center",
          ]
        }
        gap={2}
        justifyContent={
          Array [
            "",
            "end",
          ]
        }
      >
        <View__CreateCollectionMetaDetails>
          <View__CreateCollectionCuratorDetails>
            <Container
              pageType="CreateCollectionPage"
              statistics={
                Array [
                  Object {
                    "count": 0,
                    "name": "Articles",
                  },
                  Object {
                    "count": 0,
                    "name": "Collections",
                  },
                ]
              }
            />
            <Typography__component
              color="white"
            >
              Curator
            </Typography__component>
            <View__CreateCollectionCurators>
              <UserAvatarComponent
                avatar="avatar url"
                fullWidth={true}
                userId="234"
                username="345"
                variant="white"
              />
            </View__CreateCollectionCurators>
          </View__CreateCollectionCuratorDetails>
        </View__CreateCollectionMetaDetails>
      </Stack>
    </PrimaryHeaderSection>
    <View__ContentSection>
      <FormikConnect(FieldArrayInner)
        name="sections"
        render={[Function]}
      />
      <DisplayFormikState
        errors={Object {}}
        isSubmitting={false}
        touched={
          Object {
            "description": true,
            "name": true,
          }
        }
        values={
          Object {
            "name": "test name",
            "sections": Array [],
            "tags": Array [],
          }
        }
      />
    </View__ContentSection>
  </Form>
</View__Section>
`;
